radixdlt/src/main/java/com/radixdlt/consensus/safety/SafetyRules.java

```diff
@@ -41,19 +43,21 @@
 	private final ECKeyPair selfKey; // TODO remove signing/address to separate identity management
 	private final Hasher hasher;
 	private final HashSigner signer;
-
+	private final TimeSupplier timeSupplier;
 	private SafetyState state;
 
 	public SafetyRules(
 		ECKeyPair selfKey,
 		SafetyState initialState,
 		Hasher hasher,
-		HashSigner signer
+		HashSigner signer,
+		TimeSupplier timeSupplier
```

Comment:
Having the timeSupplier in SafetyRules seems to be the wrong place. It should only be in charge of ensuring safety and signing things and not need to retrieve time from another module. I think it should be given the time in the `voteFor` method.

Reply:
Yeah, that makes sense.  Will change.